lib/wanda/results.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Wanda.Results do
+  @moduledoc """
+  This module exposes functionalities to interact with the historycal log of ExecutionResults.
+  It allows to:
+   - append an item to the log
+  """
+
+  alias Wanda.Execution.{ExecutionResult, Result}
+
+  @spec append(Result.t()) :: {:ok, any()} | {:error, any()}
+  def append(
+        %Result{
+          execution_id: execution_id,
+          group_id: group_id
+        } = result
+      ) do
+    %ExecutionResult{
+      execution_id: execution_id,
+      group_id: group_id,
+      payload: result
```

Comment:
I would remove the `execution_id` and `group_id` from the payload map, otherwise they are duplicated in the main columns and the map, which looks "ugly" so to say

Reply:
@fabriziosestito The credo option to warn the one line and one parameter pipes, not enabled yet I guess XD

Reply:
@arbulu89 we thought about having them as external fields because of indexing
we could remove those two fields from the payload indeed.

Reply:
Not a blocking thing to me, but hey if you prefer so :smile: 

That would mean doing a struct to map transformation of the payload and then remove the `execution_id` and `group_id`.
The doubt I have is that `Map.from_struct` is not recursive and it will transform only the top level struct to a map (which could be enough to remove the two fields).
Are you ok with this?

Reply:
If the json encoders are there the `Map.from_struct` should work on nested things (this is actually how it is able to store the struct in the database).

This maybe would work:
```
transformed_result = result
|> Map.to_struct()
|> Map.drop([:execution_id, :group_id])
```

Reply:
@nelsonkopliku , edit:
You were right, the `Map.from_struct` doesn't map the nested structs, but as you said, execution_id and group_id are in the 1st layer, so the code snippet still would do the work I think

Reply:
Alright added in the last commit.